build.clj

```diff
@@ -67,14 +67,33 @@
   (dd/deploy {:installer :remote :artifact jar-path
               :pom-file (b/pom-path {:lib lib :class-dir class-dir})}))
 
+(defn fib [a b]
+  (lazy-seq (cons a (fib b (+ a b)))))
+
+(defn retry-with-fib-backoff [retries exec-fn test-fn]
+  (loop [idle-times (take retries (fib 1 2))]
+    (let [result (exec-fn)]
+      (if (test-fn result)
```

Comment:
I tried this out and it seems to be incorrect:

`(retry-with-fib-backoff 10 (fn [] {:failure? true}) :failure?)`
; => {:failure? true}

`(retry-with-fib-backoff 10 (fn [] (do)) :failure?)`
; prints the following:
; Returned:  nil
; Retrying with remaining back-off times (in s):  (1 2 3 5 8 13 21 34 55 89)
; ...

Reply:
thanks for the catch. same for datahike, then.